### ISSUE TITLE
connected and disconnected ports hierarchy

### ISSUE THREAD

**Issue by websterek** [jchanvfx/NodeGraphQt#130]
Connection and Disconnection signals are emitting variables with source and target port sadly it depends on the "grabbed" port. So if you try to connect the output port of node A to the input port of node B you will have different output than if you grab at first the input port of node B and connect it to the output port of node A... it's confusing so here are some screens.

![image](https://user-images.githubusercontent.com/40537787/67165199-8427a700-f382-11e9-9d0a-c825b1ca8542.png)

**Why it's a problem?**
There is no function like "get input nodes" or "get output node" so you can't get the flow of the nodes to create an operation chain. 

**Solution proposal**
Target and source should be always the same (target should be input port and source should be output port).
Or add a function to get connected nodes for input and output ports.

**Comment by mikeporetti** [comment]
vote for "function to get connected nodes for input and output ports."

**Comment by jchanvfx** [comment]
I've released a `v0.0.15` with the updated port_connected signals and also added a few new functions to the `BaseNode` class.

https://jchanvfx.github.io/NodeGraphQt/api/html/nodes.html#NodeGraphQt.BaseNode.on_input_connected

```python
from NodeGraphQt import BaseNode, NodeGraph, QtWidgets, setup_context_menu


class MyNode(BaseNode):

    # set the initial default node name.
    NODE_NAME = 'my node'

    def __init__(self):
        super(MyNode, self).__init__()
        self.add_input('foo')
        self.add_output('bar')

    # re-implementing these functions will work with the undo stack.
    def on_input_connected(self, in_port, out_port):
        print('"{}" connected to "{}:{}"'
              .format(self.name(), out_port.node().name(), out_port.name()))

    def on_input_disconnected(self, in_port, out_port):
        print('"{}" disconnected from "{}:{}"'
              .format(self.name(), out_port.node().name(), out_port.name()))


def connection_changed(in_port, out_port):
    print('hello world')


if __name__ == '__main__':
    app = QtWidgets.QApplication([])

    graph = NodeGraph()
    graph.register_node(MyNode)
    graph.widget.resize(1100, 800)
    graph.widget.show()

    setup_context_menu(graph)

    # create a couple nodes.
    node_a = graph.create_node('nodeGraphQt.nodes.MyNode',
                               name='node a',
                               pos=[0, 0])
    node_b = graph.create_node('nodeGraphQt.nodes.MyNode',
                               name='node b',
                               pos=[350, 0])

    # connect the nodes.
    node_a.set_output(0, node_b.input(0))

    # wire up connection changed func.
    graph.port_connected.connect(connection_changed)

    print(node_a.connected_output_nodes())
    print(node_b.connected_input_nodes())

    app.exec_()
```
